frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -1,41 +1,54 @@
 package bloop.engine
 
 import bloop.cli.{CliOptions, Commands, CommonOptions, ExitStatus}
+import bloop.io.SourceWatcher
 import bloop.io.Timer.timed
 import bloop.reporter.ReporterConfig
 import bloop.engine.tasks.{CompileTasks, TestTasks}
 import bloop.Project
+import bloop.logging.BloopLogger
 
 object Interpreter {
-  def execute(action: Action, state: State): State = {
-    import state.logger
+  def execute(action: Action, state0: State): State = {
+    import state0.logger
     def logAndTime[T](cliOptions: CliOptions, action: => T): T =
-      logger.verboseIf(cliOptions.verbose) { timed(state.logger) { action } }
+      logger.verboseIf(cliOptions.verbose) { timed(state0.logger) { action } }
+    def updateState(state: State, commonOptions: CommonOptions): State = {
+      State.updateLogger(state.logger, commonOptions)
+      val logger = BloopLogger(state.logger.name)
+      state.copy(commonOptions = commonOptions, logger = logger)
+    }
+
     action match {
-      case Exit(exitStatus) if state.status.isOk => state.copy(status = exitStatus)
-      case Exit(exitStatus) => state
+      case Exit(exitStatus) if state0.status.isOk => state0.copy(status = exitStatus)
```

Comment:
We need a new logger because log4j forces us to call the logger again to get the changes we just made, that involves calling `LogManager.getLogger` (which is done in the constructor of `BloopLogger`).